eogrow/core/config.py

```diff
@@ -155,14 +154,11 @@ def _recursive_apply_to_strings(config: object, function: Callable) -> object:
 
 
 def _recursive_check_config(config: object) -> None:
-    """Recursively checks if the config satisfies basic conditions for being JSON serializable.
-
-    :raises: ValueError
-    """
+    """Recursively checks if the config satisfies basic conditions for being JSON serializable."""
     if isinstance(config, dict):
         for key, value in config.items():
             if not isinstance(key, str):
-                raise ValueError(f"Config keys should be strings but {key} found")
+                raise TypeError(f"Config keys should be strings but {key} found")
```

Comment:
not sure if your description is correct :thinking:  i'd understand it as if the value has the wrong type, but it is the key that is wrong

Reply:
It's definitely confusing, so I might be wrong, but looking at the code it seems to be looping over keys and values of the config, and it checks if the keys are all strings ( I guess to avoid having numbers etc. as keys). I guess I understood correctly, but simulating the output in my head I see that it was completely fine before.